src/txn/txn_ckpt.c

```diff
@@ -405,6 +405,20 @@ __checkpoint_reduce_dirty_cache(WT_SESSION_IMPL *session)
 	 */
 	if (cache_size < 10 * WT_MEGABYTE)
 		return;
+
+	/*
+	 * Skip scrubbing if it won't perform at-least some minimum amount of
+	 * work. Scrubbing is supposed to bring down the dirty data to eviction
+	 * checkpoint target before the actual checkpoint starts. Do not perform
+	 * scrubbing if the dirty data to scrub is less than a pre-configured
+	 * size. This size is to an extent based on the configured cache size
+	 * without being too large or too small for large cache sizes.
```

Comment:
This comment does not help me understand the calculation below.  There are a lot of magic numbers in there.  Please comment on what the calculations mean and how/why it was chosen.  Is the 0.2 for 20% of the cache_size related to the default eviction_dirty_trigger setting?  Should it use that instead?

Reply:
The numbers are arbitrary to some extent. I talked to the support folks to have an idea of what the installation sizes, IO capacity people usually run with (onAWS, 100+ GB cache is common, 50 MBPS would be median IO bandwidth and tops at 320 MBPS, is what I heard). Then discussed with @agorrod and we decided to have a minimum of 500 MB or 0.2% of cache size as the delta that scrub should clean. There is no concrete reason behind that value though. I updated comment to reflect this. I am not too happy with the comment and open to suggestions on improving it.